src/node.cc

```diff
@@ -4108,6 +4135,29 @@ inline void PlatformInit() {
   }
 #endif  // !NODE_SHARED_MODE
 
+  // Record the state of the stdio file descriptors so we can restore it
+  // on exit.  Needs to happen before installing signal handlers because
+  // they make use of that information.
+  for (auto& s : stdio) {
+    const int fd = &s - stdio;
+    int err;
+
+    s.flags = GetFileDescriptorFlags(fd);
+    CHECK_NE(s.flags, -1);
+
+    do {
```

Comment:
In my opinions isatty check should be added. Something like this:
`    if( isatty(fd) ){
      do {
        err = tcgetattr(fd, &s.termios);
      } while (err == -1 && errno == EINTR);
      if (err == 0) {
        s.isatty = true;
      } else {
        // tcgetattr() is not supposed to return ENODEV or EOPNOTSUPP
        // but it does so anyway on MacOS.
        CHECK(errno == ENODEV || errno == ENOTTY || errno == EOPNOTSUPP );
      }
    }`